src/libraries/System.Security.Cryptography/tests/X509Certificates/X500DistinguishedNameBuilderTests.cs

```diff
@@ -0,0 +1,379 @@
+        public static void AddCommonName_Success_EncodeToUTF8String()
+        {
+            AssertBuilder("300D310B300906035504030C024B4A", builder => builder.AddCommonName("KJ"));
+        }
+
+        [Fact]
+        public static void AddCommonName_NullOrEmpty_Fails()
+        {
+            AssertAddThrowsOnNullAndEmpty("commonName", builder => builder.AddCommonName);
+        }
+
+        [Fact]
+        public static void AddLocalityName_Success_EncodeToUTF8String()
+        {
+            AssertBuilder("300F310D300B06035504070C04486F6D65", builder => builder.AddLocalityName("Home"));
+        }
+
+        [Fact]
+        public static void AddLocalityName_NullOrEmpty_Fails()
+        {
+            AssertAddThrowsOnNullAndEmpty("localityName", builder => builder.AddLocalityName);
+        }
+
+        [Fact]
+        public static void AddCountryOrRegion_Success_EncodeToPrintableString()
+        {
+            AssertBuilder("300D310B3009060355040613025553", builder => builder.AddCountryOrRegion("US"));
+        }
+
+        [Theory]
+        [InlineData("A")]
+        [InlineData("AAA")]
+        public static void AddCountryOrRegion_InvalidLength_Fails(string countryOrRegion)
+        {
+            X500DistinguishedNameBuilder builder = new();
+            AssertExtensions.Throws<ArgumentException>(
+                "twoLetterCode",
+                () => builder.AddCountryOrRegion(countryOrRegion));
+        }
+
+        [Fact]
+        public static void AddCountryOrRegion_InvalidContents_Fails()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            ArgumentException ex = AssertExtensions.Throws<ArgumentException>(
+                "twoLetterCode",
+                () => builder.AddCountryOrRegion("``"));
+            Assert.Contains("'PrintableString'", ex.Message);
+        }
+
+        [Fact]
+        public static void AddOrganizationName_Success_EncodeToUTF8String()
+        {
+            AssertBuilder("3011310F300D060355040A0C06476974487562", builder => builder.AddOrganizationName("GitHub"));
+        }
+
+        [Fact]
+        public static void AddOrganizationName_NullOrEmpty_Fails()
+        {
+            AssertAddThrowsOnNullAndEmpty("organizationName", builder => builder.AddOrganizationName);
+        }
+
+        [Fact]
+        public static void AddOrganizationalUnitName_Success_EncodeToUTF8String()
+        {
+            AssertBuilder("300E310C300A060355040B0C03505345", builder => builder.AddOrganizationalUnitName("PSE"));
+        }
+
+        [Fact]
+        public static void AddOrganizationalUnitName_NullOrEmpty_Fails()
+        {
+            AssertAddThrowsOnNullAndEmpty("organizationalUnitName", builder => builder.AddOrganizationalUnitName);
+        }
+
+        [Fact]
+        public static void AddStateOrProvinceName_Success_EncodeToUTF8String()
+        {
+            AssertBuilder("300D310B300906035504080C024341", builder => builder.AddStateOrProvinceName("CA"));
```

Comment:
I certainly don't want to make this a negative test, but, IIRC, it's supposed to be the full name of the state, not the abbreviation.  So "California"

> 6.3.5 State or Province Name
The State or Province Name attribute type specifies a state or province. When used as a component of a directory name, it identifies a geographical subdivision in which the named object is physically located or with which it is associated in some other important way.
An attribute value for State or Province Name is a string, e.g., **S = "Ohio"**.
stateOrProvinceName ATTRIBUTE ::= {
SUBTYPE OF name
WITH SYNTAX UnboundedDirectoryString
LDAP-SYNTAX directoryString.&id
LDAP-NAME {"st"}
ID id-at-stateOrProvinceName }

(S = Ohio, emphasis mine)